packages/flutter_tools/lib/src/web/chrome.dart

```diff
@@ -235,7 +235,17 @@ class ChromiumLauncher {
     ), skipCheck);
   }
 
-  Future<Process?> _spawnChromiumProcess(List<String> args) async {
+  Future<Process?> _spawnChromiumProcess(List<String> args, String chromeExecutable) async {
+    if (_operatingSystemUtils.hostPlatform == HostPlatform.darwin_arm) {
+      final ProcessResult result = _processManager.runSync(<String>['file', chromeExecutable]);
+      // Check if ARM Chrome is installed.
+      // Mach-O 64-bit executable arm64
+      if ((result.stdout as String).contains('arm64')) {
```

Comment:
should we leave ourselves a `printTrace()` breadcrumb here?

Reply:
Done.